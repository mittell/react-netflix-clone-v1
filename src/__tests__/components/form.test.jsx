/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/prefer-screen-queries */
import React from 'react';
import { render } from '@testing-library/react';
import { Form } from '../../components';

jest.mock('react-router-dom');

describe('<Form/>', () => {
	it('renders <Form/> with populated data', () => {
		const { container, getByText, getByPlaceholderText } = render(
			<Form>
				<Form.Title>Sign In Now</Form.Title>

				<Form.Base>
					<Form.Input placeholder='Email Address' onChange={() => {}} />
					<Form.Input
						type='password'
						placeholder='Password'
						onChange={() => {}}
					/>
					<Form.Submit disabled type='submit'>
						Sign In
					</Form.Submit>
				</Form.Base>

				<Form.Text>
					New to Aniflix? <Form.Link to='/signup'>Sign up now</Form.Link>.
				</Form.Text>
				<Form.TextSmall>
					This page is protected by Google reCAPTCHA to ensure you're not a bit.
					Learn more.
				</Form.TextSmall>
			</Form>
		);

		expect(
			getByText(
				"This page is protected by Google reCAPTCHA to ensure you're not a bit. Learn more."
			)
		).toBeTruthy();
		expect(getByText('Sign In Now')).toBeTruthy();
		expect(getByText('Sign In')).toBeTruthy();
		expect(getByText('Sign In').disabled).toBeTruthy();
		expect(getByPlaceholderText('Email Address')).toBeTruthy();
		expect(getByPlaceholderText('Password')).toBeTruthy();
		expect(container.firstChild).toMatchSnapshot();
	});

	it('renders <Form/> with an error', () => {
		const { container, getByText, queryByText } = render(
			<Form>
				<Form.Error>Your email address is already being used</Form.Error>
				<Form.Submit type='submit'>Sign In</Form.Submit>
			</Form>
		);

		expect(getByText('Your email address is already being used')).toBeTruthy();
		expect(getByText('Sign In')).toBeTruthy();
		expect(queryByText('Sign In').disabled).toBeFalsy();
		expect(container.firstChild).toMatchSnapshot();
	});
});
