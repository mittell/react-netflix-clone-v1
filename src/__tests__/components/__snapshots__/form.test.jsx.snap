// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> renders <Form/> with an error 1`] = `
<div
  class="form__Container-sc-7erizj-0 esGAAj"
>
  <div
    class="form__Error-sc-7erizj-1 bYcoco"
  >
    Your email address is already being used
  </div>
  <button
    class="form__Submit-sc-7erizj-8 koFFBU"
    type="submit"
  >
    Sign In
  </button>
</div>
`;

exports[`<Form/> renders <Form/> with populated data 1`] = `
<div
  class="form__Container-sc-7erizj-0 esGAAj"
>
  <h1
    class="form__Title-sc-7erizj-3 bBsRdF"
  >
    Sign In Now
  </h1>
  <form
    class="form__Base-sc-7erizj-2 gEaooO"
  >
    <input
      class="form__Input-sc-7erizj-7 bOiiFd"
      placeholder="Email Address"
    />
    <input
      class="form__Input-sc-7erizj-7 bOiiFd"
      placeholder="Password"
      type="password"
    />
    <button
      class="form__Submit-sc-7erizj-8 koFFBU"
      disabled=""
      type="submit"
    >
      Sign In
    </button>
  </form>
  <p
    class="form__Text-sc-7erizj-4 epXxZU"
  >
    New to Aniflix? 
    .
  </p>
  <p
    class="form__TextSmall-sc-7erizj-5 hbTGBU"
  >
    This page is protected by Google reCAPTCHA to ensure you're not a bit. Learn more.
  </p>
</div>
`;
